src/EnergyPlus/HeatingCoils.cc

```diff
@@ -3320,6 +3340,74 @@ namespace HeatingCoils {
 
 	}
 
+	void
+	SetHeatingCoilAsDesicRegenCoil(
+		std::string const & CoilType, // must match coil types in this module
+		std::string const & CoilName, // must match coil names for the coil type
+		int & DesiccantDehumIndex, // index of desiccant dehumidifier
+		bool & ErrorsFound // set to true if problem
+		) {
+
+		// FUNCTION INFORMATION:
+		//       AUTHOR         Bereket Nigusse
+		//       DATE WRITTEN   February 2016
+		//       MODIFIED       na
+		//       RE-ENGINEERED  na
+
+		// PURPOSE OF THIS FUNCTION:
+		// This function looks up the given coil and registers the coil as desiccant regeneration air heating coil.
+		// If incorrect coil type or name is given, ErrorsFound is returned as true
+
+		// METHODOLOGY EMPLOYED:
+		// na
+
+		// REFERENCES:
+		// na
+
+		// Using/Aliasing
+		using InputProcessor::FindItem;
+
+		// Return value
+		// na
+
+		// Locals
+		// FUNCTION ARGUMENT DEFINITIONS:
+
+		// FUNCTION PARAMETER DEFINITIONS:
+		// na
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+		// na
+
+		// DERIVED TYPE DEFINITIONS:
+		// na
+
+		// FUNCTION LOCAL VARIABLE DECLARATIONS:
+		int WhichCoil;
+		int FoundType; // Integer equivalent of coil type
+
+		// Obtains and Allocates HeatingCoil related parameters from input file
+		if ( GetCoilsInputFlag ) { //First time subroutine has been entered
+			GetHeatingCoilInput();
+			GetCoilsInputFlag = false;
+		}
+
+		WhichCoil = 0;
+		FoundType = FindItem( CoilType, cAllCoilTypes, NumAllCoilTypes );
```

Comment:
@Nigusse @rraustad Why is this function searching by type and name?  The dehumidifier should know the index (WhichCoil).  This should be a 2 line function.  I see it's following the pattern of GetHeatingCoilInletNode and GetHeatingCoilOutletNode and GetHeatingCoilControlNodeNum.  Doesn't seem like those function should be searching by name either.  I know they are used in other places, but seems that they could all require using GetHeatingCoilIndex first which the parent object probably has already stored by then, then simply pass the index to these other functions.